nx-X11/programs/Xserver/hw/nxagent/Reconnect.c

```diff
@@ -438,22 +438,10 @@ Bool nxagentReconnectSession(void)
 
   if (nxagentKeyboard != NULL)
   {
-    int size;
+    nxagentOldKeyboard = strndup(nxagentKeyboard, strlen(nxagentKeyboard));
+    /* FIXME: check if nxagentOldKeyboard == NULL and abort */
```

Comment:
nxagentOldKeyboard is a local var. And it gets set in the code section above. nxagentKeyboard comes in via the cmdline (see Args.c). It indeed never gets properly sanitized. But that is something for another PR. And it should be done in Args.c, right?